src/Pipeline.h

```diff
@@ -376,16 +413,19 @@ class Pipeline {
      * shape, but the shape can vary across the different output
      * Funcs. This form of realize does *not* automatically copy data
      * back from the GPU. */
-    void realize(Realization dst, const Target &target = Target(), const ParamMap &param_map = ParamMap());
+    NO_INLINE void realize(RealizationArg output, const Target &target = Target(), 
+                           const ParamMap &param_map = ParamMap::empty_map());
 
     /** For a given size of output, or a given set of output buffers,
      * determine the bounds required of all unbound ImageParams
      * referenced. Communicates the result by allocating new buffers
      * of the appropriate size and binding them to the unbound
      * ImageParams. */
     // @{
-    void infer_input_bounds(int x_size = 0, int y_size = 0, int z_size = 0, int w_size = 0, const ParamMap &param_map = ParamMap());
-    void infer_input_bounds(Realization dst, const ParamMap &param_map = ParamMap());
+    void infer_input_bounds(int x_size = 0, int y_size = 0, int z_size = 0, int w_size = 0, 
+                            const ParamMap &param_map = ParamMap::empty_map());
+    NO_INLINE void infer_input_bounds(RealizationArg output,
```

Comment:
NO_INLINE unnecessary

The purpose of NO_INLINE is to prevent code in the Halide namespace that might throw an error from being inlined into the user's code, because then the line number where the error occurred can't be accurately reported.